app/src/main/java/com/android/unio/model/search/SearchViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.android.unio.model.search
+
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.android.unio.model.association.Association
+import com.android.unio.model.association.AssociationRepository
+import com.android.unio.model.event.Event
+import com.android.unio.model.event.EventRepository
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for searching associations and events. It uses a [SearchRepository] to create the
+ * AppSearch database and exposes the results through a [StateFlow] containing a list of
+ * respectively [Association] and [Event]
+ */
+class SearchViewModel(private val repository: SearchRepository) : ViewModel() {
+  private val _associations = MutableStateFlow<List<Association>>(emptyList())
+  val associations: StateFlow<List<Association>> = _associations
+
+  private val _events = MutableStateFlow<List<Event>>(emptyList())
+  val events: StateFlow<List<Event>> = _events
+
+  /** Initializes the ViewModel by creating the search database and connecting it to the session. */
+  init {
+    Firebase.auth.addAuthStateListener {
+      if (it.currentUser != null) {
+        viewModelScope.launch { repository.init() }
+      }
```

Comment:
Have a look at how the `init` block is defined in `AssociationViewModel`:
```kotlin
init {
    associationRepository.init { getAssociations() }
  }
``` 

It simply calls the repository's `init` function. You'll see that the addition of the Auth State Listener is done in that repository `init` function, e.g. in `AssociationRepositoryFirestore`:
```kotlin
override fun init(onSuccess: () -> Unit) {
    Firebase.auth.addAuthStateListener {
      if (it.currentUser != null) {
        onSuccess()
      }
    }
  }
``` 

I think it makes more sense to add the Auth State Listener there as it is the database's duty, but let me know if you disagree